server/server.go

```diff
@@ -747,6 +749,29 @@ func (a *ArgoCDServer) newHTTPServer(ctx context.Context, port int, grpcWebHandl
 		handler = compressHandler(handler)
 	}
 	mux.Handle("/api/", handler)
+	terminalHandler := terminal.NewHandler(a.appLister, a.db, a.enf, a.Cache, appResourceTreeFn)
+	mux.HandleFunc("/terminal", func(writer http.ResponseWriter, request *http.Request) {
+		if !a.DisableAuth {
+			ctx := request.Context()
+			cookies := request.Cookies()
+			tokenString, _ := httputil.JoinCookies(common.AuthCookieName, cookies)
```

Comment:
Is this call necessarily-safe, or should we handle the error?

Reply:
Thanks for catching this. Just added some logic to handle this error.